File: frontend/src/schemas/tournament.ts
import { z } from "zod";

export const tournamentSchema = z
    .object({
        name: z
            .string()
            .min(3, "Минимум 3 символа")
            .max(100, "Максимум 100 символов"),
        description: z.string().optional(),
        type: z.enum([
            "single_elimination",
            "double_elimination",
            "round_robin",
        ]),
        rules: z.string().optional(),
        max_teams: z
            .number()
            .min(2, "Минимум 2 команды")
            .max(64, "Максимум 64 команды"),
        registration_deadline: z.string().datetime(),
        start_date: z.string().datetime(),
        end_date: z.string().datetime(),
    })
    .refine(
        (data) =>
            new Date(data.registration_deadline) < new Date(data.start_date),
        {
            message: "Дедлайн регистрации должен быть раньше даты начала",
            path: ["registration_deadline"],
        }
    )
    .refine((data) => new Date(data.start_date) < new Date(data.end_date), {
        message: "Дата начала должна быть раньше даты окончания",
        path: ["start_date"],
    });

export type TournamentFormData = z.infer<typeof tournamentSchema>;

File: backend/app/models/tournament.py
import enum
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, Table, Enum
from app.db.base_class import Base
from datetime import datetime

class TournamentStatus(str, enum.Enum):
    DRAFT = "DRAFT"
    REGISTRATION = "REGISTRATION"
    IN_PROGRESS = "IN_PROGRESS"
    COMPLETED = "COMPLETED"

class Tournament(Base):
    __tablename__ = "tournaments"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, index=True)
    description = Column(String, nullable=True)
    type = Column(String)  # single_elimination, double_elimination, round_robin
    rules = Column(String, nullable=True)
    max_teams = Column(Integer, nullable=True)
    start_date = Column(DateTime, nullable=True)
    end_date = Column(DateTime, nullable=True)
    created_by = Column(Integer, ForeignKey("users.id"))
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    status = Column(Enum(TournamentStatus), default=TournamentStatus.REGISTRATION)

    # ... остальные отношения ... 